src/pages/iou/steps/NewRequestAmountPage.js

```diff
@@ -0,0 +1,242 @@
+    };
+
+    const navigateToNextPage = (currentAmount) => {
+        const amountInSmallestCurrencyUnits = CurrencyUtils.convertToSmallestUnit(currency, Number.parseFloat(currentAmount));
+        IOU.setMoneyRequestAmount(amountInSmallestCurrencyUnits);
+        IOU.setMoneyRequestCurrency(currency);
+
+        if (isEditing) {
+            Navigation.goBack(ROUTES.getMoneyRequestConfirmationRoute(iouType, reportID));
+            return;
+        }
+
+        const moneyRequestID = `${iouType}${reportID}`;
+        const shouldReset = iou.id !== moneyRequestID;
+        // If the money request ID in Onyx does not match the ID from params, we want to start a new request
+        // with the ID from params. We need to clear the participants in case the new request is initiated from FAB.
+        if (shouldReset) {
+            IOU.setMoneyRequestId(moneyRequestID);
+            IOU.setMoneyRequestDescription('');
+            IOU.setMoneyRequestParticipants([]);
+        }
+
+        // If a request is initiated on a report, skip the participants selection step and navigate to the confirmation page.
+        if (report.reportID) {
+            // Reinitialize the participants when the money request ID in Onyx does not match the ID from params
+            if (_.isEmpty(participants)) {
+                const currentUserAccountID = currentUserPersonalDetails.accountID;
+                const iouParticipants = ReportUtils.isPolicyExpenseChat(report)
+                    ? [{reportID: report.reportID, isPolicyExpenseChat: true, selected: true}]
+                    : _.chain(report.participantAccountIDs)
+                          .filter((accountID) => currentUserAccountID !== accountID)
+                          .map((accountID) => ({accountID, selected: true}))
+                          .value();
+                IOU.setMoneyRequestParticipants(iouParticipants);
+            }
+            Navigation.navigate(ROUTES.getMoneyRequestConfirmationRoute(iouType, reportID));
+            return;
+        }
+        Navigation.navigate(ROUTES.getMoneyRequestParticipantsRoute(iouType));
+    };
+
+    const content = (
+        <MoneyRequestAmountForm
+            isEditing={isEditing}
+            currency={currency}
+            amount={amount}
+            ref={(e) => (textInput.current = e)}
+            onCurrencyButtonPress={navigateToCurrencySelectionPage}
+            onSubmitButtonPress={navigateToNextPage}
+        />
+    );
+
+    // ScreenWrapper is only needed in edit mode because we have a dedicated route for the edit amount page (MoneyRequestEditAmountPage).
+    // The rest of the cases this component is rendered through <MoneyRequestSelectorPage /> which has it's own ScreenWrapper
+    if (!isEditing) {
+        return content;
+    }
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            onEntryTransitionEnd={focusTextInput}
```

Comment:
NAB I think we want to split this into two pages, one that uses `ScreenWrapper` for `MoneyRequestEditAmountPage` screen. And the other that does not (meant to be a part of the page) for `MoneyRequestSelectorPage`.

I'm questioning this change as the conditional rendering above just seems easier :grin:

Reply:
TBH I'm not sure if I can visualise this change right now 😅 Maybe this could be done later on?